lib/chef/exceptions.rb

```diff
@@ -97,6 +97,7 @@ class GroupIDNotFound < ArgumentError; end
     class ConflictingMembersInGroup < ArgumentError; end
     class InvalidResourceReference < RuntimeError; end
     class ResourceNotFound < RuntimeError; end
+    class ProviderNotFound < ArgumentError; end
```

Comment:
You're creating a new class because `NoProviderAvailable` because you want to preserve the parent class?